Commonalities/documentation/Working/user-consent.md

```diff
@@ -0,0 +1,396 @@
+
+The Purpose Levels of the table below are direct translations from the [Art.6 of the GDPR](https://gdpr-info.eu/art-6-gdpr/). For example, this sentence from that article: _The data subject has given consent to the processing of his or her personal data for one or more specific purpose._
+
+Shall be translated into this:  _In order to process a specific personal information resource, the user needs to give his/her consent to a purpose which has a purpose level of type "consent"_.
+
+
+| **Purpose Level** | Description | Type |
+| --- | --- | --- |
+|`legal_obligation` | A process that is required by law. | Automatic purpose |
+|`contract` | A process that is needed to fulfill the service contract with the user, and it is already approved when the contract was signed. | Automatic purpose |
+|`legitimate_interest`| An interest of the data processor that does not conflict with the fundamental interest of the individuals. | Opt-out purpose |
+|`compatible` | A re-processing of data acquired through another purpose that is compatible with the first purpose. | Opt-out purpose|
+|`consent` | This level applies to purposes which are unrelated enough to the service. | Opt-in purpose|
+|`terms_and_conditions`| The operations or tasks done by an application to provide its functionality, as described in the Terms and Conditions. | Opt-in purpose|
+
+
+> _NOTE: This list of levels MAY be extended with new values. The OpenID Provider (Auth Server) and the APIs used by the Relying Parties (client Applications) MUST be ready to support new values in the future._
+
+Notice that `terms_and_conditions` is not mentioned in GDPR law, but as it obeys to the same rules that a _consent_ (it requires the user opt-in), it has been modelled in that way.
+
+### Purpose relationship with other key concepts
+
+The key concepts defined to handle user personal information are related to purpose concept in the following way:
+- Scopes tell what resources (e.g. APIs) are accessible under those scopes.
+- A purpose has associated a set of PI scopes. 
+  - These are scopes which refer to resources providing access to PI and thus requiring special protection as stated by the GDPR law.
+  - Getting access to PI scopes MUST always be done by explicitly declaring a purpose.
+- Scopes are assigned to an application, granting access to APIs and other resources.
+- Purposes are assigned to an application.
+
+Therefore, an application will have access to:
+  - APIs or resources covered by the assigned scopes
+  - APIs or resources providing personal information under the assigned purpose (because of the PI scopes included in that purpose)
+    - **Only** if user has granted her consent for that purpose
+    - Which may happen automatically (automatic purposes) or may need a user opt-in depending on the purpose level of the purpose associated to that consent. In the second case, the consent is created and stored through Operator NaaS platform.
```

Comment:
- Would be good to add the picture provided in the presentation explaining the relationships here.
- Here it is mentioned that consent creation and storing is done through the operator platform. In the presentation material, under high level journey example, why does the marketplace need access to user consent management service of the operator, if operator identifies user based on identifier & uses backchannel to get his consent?

Reply:
> Would be good to add the picture provided in the presentation explaining the relationships here.

Sure. No problem.

> Here it is mentioned that consent creation and storing is done through the operator platform. In the presentation material, under high level journey example, why does the marketplace need access to user consent management service of the operator? If a marketplace/app is using our (user) consent mgmt service/api to register a user's consent, how can this be trusted?

User consent collection is delegated to the third party (data processor). The third party is responsible of capturing that consent (which is something that could be legally agreed with the third party by contract) and register it using Consent API. You are not simply relying on the third party to collect the user consent but also you request to register it on Operator side (data controller). And it should be done, IF and only IF the user has actually granted the consent. That process is logged and audited in case there is any problem.

Reply:
@jpengar Thanks for the clarification. I found this part " but also you request to register it on Operator side (data controller)" missing in the presentation picture and hence the question.
The current PR is still using auth code grant flow in its example. Will this be updated to the new proposed CIBA flow?

Reply:
Yes, we are working on it .

Reply:
@shilpa-padgaonkar It is updated already.